src/tests/promiseUtils_tests.ts

```diff
@@ -0,0 +1,106 @@
+	let done = assert.async();
+	PromiseUtils.execWithRetry(fnGenerator(2)).then((retVal) => {
+		strictEqual(retVal, expectedResolve, "execWithRetry should resolve with the same value the function's promise resolves with");
+		strictEqual(callCount, 2, "execWithRetry should call the function twice");
+	}).catch(() => {
+		ok(false, "execWithRetry should not reject");
+	}).then(() => {
+		done();
+	});
+});
+
+test("execWithRetry should reject if all attempts fail and there are not enough retries", (assert: QUnitAssert) => {
+	let done = assert.async();
+	PromiseUtils.execWithRetry(fnGenerator(Infinity)).then((retVal) => {
+		ok(false, "execWithRetry should not resolve");
+	}).catch((retVal) => {
+		strictEqual(retVal, expectedReject, "execWithRetry should reject with the same value the function's promise rejects with");
+		strictEqual(callCount, 3, "execWithRetry should call the function 3 times (1 first attempt, then 2 retries)");
+	}).then(() => {
+		done();
+	});
+});
+
+test("execWithRetry should call the function once if the first attempt succeeds, and there are 0 retries", (assert: QUnitAssert) => {
+	let done = assert.async();
+	let retryOptions: RetryOptions = { retryCount: 0, minTimeout: 0, maxTimeout: 0 };
+
+	PromiseUtils.execWithRetry(fnGenerator(1), retryOptions).then((retVal) => {
+		strictEqual(retVal, expectedResolve, "execWithRetry should resolve with the same value the function's promise resolves with");
+		strictEqual(callCount, 1, "execWithRetry should call the function once");
+	}).catch(() => {
+		ok(false, "execWithRetry should not reject");
+	}).then(() => {
+		done();
+	});
+});
+
+test("execWithRetry should call the function once then reject if the first attempt fails, and there are 0 retries", (assert: QUnitAssert) => {
+	let done = assert.async();
+	let retryOptions: RetryOptions = { retryCount: 0, minTimeout: 0, maxTimeout: 0 };
+
+	PromiseUtils.execWithRetry(fnGenerator(Infinity), retryOptions).then((retVal) => {
+		ok(false, "execWithRetry should not resolve");
+	}).catch((retVal) => {
+		strictEqual(retVal, expectedReject, "execWithRetry should reject with the same value the function's promise rejects with");
+		strictEqual(callCount, 1, "execWithRetry should call the function once");
+	}).then(() => {
+		done();
+	});
+});
+
+test("execWithRetry should call the function n times if only the nth attempt succeeds, and there are n-1 retries", (assert: QUnitAssert) => {
+	let done = assert.async();
+	let retryOptions: RetryOptions = { retryCount: 49, minTimeout: 1, maxTimeout: 2 };
+
+	PromiseUtils.execWithRetry(fnGenerator(50), retryOptions).then((retVal) => {
+		strictEqual(retVal, expectedResolve, "execWithRetry should resolve with the same value the function's promise resolves with");
+		strictEqual(callCount, 50, "execWithRetry should call the function 500 times");
+	}).catch(() => {
+		ok(false, "execWithRetry should not reject");
+	}).then(() => {
+		done();
+	});
```

Comment:
We decided that this will be low prio as we're pretty confident that the timeouts work the way they should.